nyaa/search.py

```diff
@@ -3,15 +3,99 @@
 import math
 import json
 import shlex
+from datetime import datetime, timedelta
 
 from nyaa import app, db
-from nyaa import models
+from nyaa import models, torrents
 
 import sqlalchemy_fulltext.modes as FullTextMode
 from sqlalchemy_fulltext import FullTextSearch
 from elasticsearch import Elasticsearch
 from elasticsearch_dsl import Search, Q
 
+class ElasticTorrentStatsAccessor:    
+    def __init__(self, elasticResult):
+        self.seed_count = elasticResult.seed_count
+        self.leech_count = elasticResult.leech_count
+        self.download_count = elasticResult.download_count
+
+class ElasticTorrentAccessor:
+    knownAttributes = ['id', 'has_torrent', 'display_name', 'filesize', 'anonymous', 'trusted', 'remake', 'complete', 'hidden', 'deleted']
+
+    def __init__(self, elasticResult):
+        self.elastic = elasticResult
+
+    def __getattr__(self, name):
+        # Passthru for attributes that are the same
+        if name in self.knownAttributes:
+            return self.elastic[name]
+
+        # Safetynet: throw if an attribute is accessed which we don't implement    
+        raise Exception("Unknown attribute {}".format(name))
```

Comment:
Well if you want a more defensive version, this would raise for undesired attributes:
```python
def __getattr__(self, item):
    if item not in supported_attributes:
        raise AttributeError(
            '{class_name!r} object does not support attribute {attr!r}'.format(
            class_name=self.__class__.__name__,
            attr=item,
        )
    else:
        try:
            return self.elastic[item]
        except KeyError:
            # Raise an AttributeError if the key does not exist
            raise AttributeError(
                '{class_name!r} object has no attribute {attr!r}'.format(
                    class_name=self.__class__.__name__,
                    attr=item,
                )
            )
```
But personally I'd prefer the following which raises a warning on unsupported attribute but allows execution to continue, or use logging instead and log each unsupported attribute access.
```python
import warnings

def __getattr__(self, item):
    try:
        result = self.elastic[item]
    except KeyError:
        # Raise an AttributeError if the key does not exist
        raise AttributeError(
            '{class_name!r} object has no attribute {attr!r}'.format(
                class_name=self.__class__.__name__,
                attr=item,
            )
        )
    else:
        if item not in self.known_attributes:
            warnings.warn('{class_name!r} object accessing unsupported attribute {attr!r}', AttributeError)
        return result
```